mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulPrepareGameScreen.kt

```diff
@@ -3,32 +3,89 @@ package ch.epfl.sdp.mobile.state
+import ch.epfl.sdp.mobile.application.chess.ChessFacade
+import ch.epfl.sdp.mobile.application.chess.Match
 import ch.epfl.sdp.mobile.ui.prepare_game.PrepareGameScreen
 import ch.epfl.sdp.mobile.ui.prepare_game.PrepareGameScreenState
-
-/**
- * An implementation of the [PrepareGameScreenState]
- * @property user authenticated user
- * @property colorChoice chosen color side
- * @property gameType chosen game type
- */
-class PrepareGameScreenStateImpl(private val user: AuthenticatedUser) : PrepareGameScreenState {
-  override var colorChoice: ColorChoice by mutableStateOf(ColorChoice.White)
-  override var gameType: GameType by mutableStateOf(GameType.Local)
-}
+import ch.epfl.sdp.mobile.ui.prepare_game.PrepareGameScreenState.ColorChoice
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 
 /**
  * A composable that makes a [PrepareGameScreen] stateful
  * @param user authenticated user
+ * @param navigateToGame The action to take for navigating to a created game
+ * @param cancelClick The action to take when clicking on the dialog's cancel button
  * @param modifier [Modifier] for this composable
  */
 @Composable
 fun StatefulPrepareGameScreen(
     user: AuthenticatedUser,
+    navigateToGame: (match: Match) -> Unit,
+    cancelClick: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
-  val state = remember(user) { PrepareGameScreenStateImpl(user) }
-  PrepareGameScreen(state, modifier)
+  val scope = rememberCoroutineScope()
+  val chessFacade = LocalChessFacade.current
+  val opponents =
+      remember(user) { user.following }.collectAsState(emptyList()).value.map { ProfileAdapter(it) }
+
+  val state =
+      remember(user, opponents, navigateToGame, cancelClick, scope, chessFacade) {
+        SnapshotPrepareGameScreenState(
+            user = user,
+            opponents = opponents,
+            navigateToGame = navigateToGame,
+            cancelClick = cancelClick,
+            chessFacade = chessFacade,
+            scope = scope,
+        )
+      }
+  PrepareGameScreen(state = state, modifier = modifier, key = { it.uid })
+}
+
+/**
+ * An implementation of the [PrepareGameScreenState]
+ *
+ * @param user The current [AuthenticatedUser]
+ * @param navigateToGame The action to take for navigating to a created game
+ * @param cancelClick The action to take when clicking on the dialog's cancel button
+ * @param chessFacade The chess facade to act on the store
+ * @param scope The [Composable]'s [CoroutineScope]
+ */
+class SnapshotPrepareGameScreenState(
+    val user: AuthenticatedUser,
+    override val opponents: List<ProfileAdapter>,
+    val navigateToGame: (match: Match) -> Unit,
+    val cancelClick: () -> Unit,
+    val chessFacade: ChessFacade,
+    val scope: CoroutineScope,
+) : PrepareGameScreenState<ProfileAdapter> {
+
+  override var colorChoice: ColorChoice by mutableStateOf(ColorChoice.White)
+
+  override var selectedOpponent: ProfileAdapter? by mutableStateOf(null)
```

Comment:
Mark the setter as private so a caller of `SnapshotPrepareGameScreenState` can not set its `selectedOpponent`.

```suggestion
  override var selectedOpponent: ProfileAdapter? by mutableStateOf(null)
    private set
```

(the formatting may be broken so I suggest doing the change manually)

Reply:
Fixed in [a591170](https://github.com/epfl-SDP/android/pull/200/commits/a591170a11c23c0c88ac1eb5f2f7ad367316a3f8)